extensions/amp-render/1.0/amp-render.js

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {BaseElement} from './base-element';
+import {Services} from '../../../src/services';
+import {batchFetchJsonFor} from '../../../src/batched-json';
+import {dev, user, userAssert} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {isExperimentOn} from '../../../src/experiments';
+
+/** @const {string} */
+const TAG = 'amp-render';
+
+const AMP_STATE_URI_SCHEME = 'amp-state:';
+const AMP_SCRIPT_URI_SCHEME = 'amp-script:';
+
+class AmpRender extends BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?../../../src/service/template-impl.Templates} */
+    this.templates_ = null;
+
+    /** @private {?Element} */
+    this.template_ = null;
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    userAssert(
+      isExperimentOn(this.win, 'amp-render'),
+      'Experiment "amp-render" is not turned on.'
+    );
+    return super.isLayoutSupported(layout);
+  }
+
+  /** @override */
+  init() {
+    return dict({
+      'fetchFn': this.getFetchFn_(),
+    });
+  }
+
+  /**
+   * Returns true if element's src points to amp-state.
+   *
+   * @param {string} src
+   * @return {boolean}
+   * @private
+   */
+  isAmpStateSrc_(src) {
+    return src.startsWith(AMP_STATE_URI_SCHEME);
+  }
+
+  /**
+   * Returns true if element's src points to an amp-script function.
+   *
+   * @param {string} src
+   * @return {boolean}
+   * @private
+   */
+  isAmpScriptSrc_(src) {
+    return src.startsWith(AMP_SCRIPT_URI_SCHEME);
+  }
+
+  /**
+   * Returns the correct fetch function for amp-state, amp-script or
+   * to fetch remote JSON.
+   *
+   * @return {Function}
+   * @private
+   */
+  getFetchFn_() {
+    const src = this.element.getAttribute('src');
+    if (this.isAmpStateSrc_(src)) {
+      return this.getAmpStateJson.bind(null, this.element);
+    }
+    if (this.isAmpScriptSrc_(src)) {
+      // TODO(dmanek): implement this
+      return;
```

Comment:
Let's have this return noop in the mean time so it still returns a function type
```suggestion
      return () => {};
```